lib/media-mgmt/dailyMediaManager.ts

```diff
@@ -279,59 +307,67 @@ export class DailyMediaManager extends MediaManager {
+  protected async handleTrackStarted(event: DailyEventObjectTrack) {
     if (!event.participant?.local) return;
     if (event.track.kind === "audio") {
-      const status = this._mediaStreamRecorder.getStatus();
-      switch (status) {
-        case "ended":
-          await this._mediaStreamRecorder.begin(event.track);
-          if (this._connected) {
-            this._startRecording();
-            if (this._connectResolve) {
-              this._connectResolve();
-              this._connectResolve = null;
-            }
-          }
-          break;
-        case "paused":
-          this._startRecording();
-          break;
-        case "recording":
-        default:
-          if (this._currentAudioTrack !== event.track) {
-            await this._mediaStreamRecorder.end();
+      if (this._mediaStreamRecorder) {
+        const status = this._mediaStreamRecorder.getStatus();
+        switch (status) {
+          case "ended":
             await this._mediaStreamRecorder.begin(event.track);
+            if (this._connected) {
+              this._startRecording();
+              if (this._connectResolve) {
+                this._connectResolve();
+                this._connectResolve = null;
+              }
+            }
+            break;
+          case "paused":
             this._startRecording();
-          } else {
-            console.warn(
-              "track-started event received for current track and already recording"
-            );
-          }
-          break;
+            break;
+          case "recording":
+          default:
+            if (this._currentAudioTrack !== event.track) {
+              await this._mediaStreamRecorder.end();
+              await this._mediaStreamRecorder.begin(event.track);
+              this._startRecording();
+            } else {
+              console.warn(
+                "track-started event received for current track and already recording",
+              );
+            }
+            break;
+        }
       }
-      this._callbacks.onTrackStarted?.(
-        event.track,
-        event.participant
-          ? dailyParticipantToParticipant(event.participant)
-          : undefined
-      );
+      this._currentAudioTrack = event.track;
     }
+    this._callbacks.onTrackStarted?.(
+      event.track,
+      event.participant
+        ? dailyParticipantToParticipant(event.participant)
+        : undefined,
+    );
+    this.onTrackStartedCallback?.(event);
   }
 
-  private _handleTrackStopped(event: DailyEventObjectTrack) {
+  protected handleTrackStopped(event: DailyEventObjectTrack) {
     if (!event.participant?.local) return;
     if (event.track.kind === "audio") {
-      if (this._mediaStreamRecorder.getStatus() === "recording") {
+      if (
+        this._mediaStreamRecorder &&
+        this._mediaStreamRecorder.getStatus() === "recording"
+      ) {
         this._mediaStreamRecorder.pause();
       }
       this._callbacks.onTrackStopped?.(
```

Comment:
should this callback be moved outside the `if audio` check to happen for video too?

Reply:
Yep. I thought I had already done that. But I was wrong. 😅

Reply:
Done.